BenchPress/Helpers/BenchPress.Azure/Common.psm1

```diff
@@ -121,19 +126,202 @@ function Get-ResourceByType {
+    }
+  }
+  return $ConfirmResult
+}
+
+
+<#
+  .SYNOPSIS
+    Private function to create a message and ErrorRecord for when a resource does not exist.
+
+  .DESCRIPTION
+    Format-NotExistError is a private helper function that can be used to construct a message and ErrorRecord
+    for when a resource does not exist.
+
+  .PARAMETER Expected
+    The name of the resource that was expected to exist.
+
+  .EXAMPLE
+    Format-NotExistError -Expected "MyVM"
+
+  .INPUTS
+    System.String
+
+  .OUTPUTS
+    System.Management.Automation.ErrorRecord
+#>
+function Format-NotExistError([string]$Expected) {
+  $Message = "Expected $Expected to exist, but it does not exist."
+  return Format-ErrorRecord -Message $Message -ErrorID "BenchPressExistFail"
+}
+
+<#
+  .SYNOPSIS
+    Private function to create a message and ErrorRecord for when a resource property is not set correctly.
+
+  .DESCRIPTION
+    Format-IncorrectValueError is a private helper function that can be used to construct a message and ErrorRecord
+    for when a resource property is not set to the expected value.
+
+  .PARAMETER ExpectedKey
+    The resource property name that is checked
+
+  .PARAMETER ExpectedValue
+    The expected value of the resource property
+
+  .PARAMETER ActualValue
+    The actual value of the resource property
+
+  .EXAMPLE
+    Format-IncorrectValueError -ExpectedKey "Location" -ExpectedValue "WestUS3" -ActualValue "EastUS"
+
+  .INPUTS
+    System.String
+
+  .OUTPUTS
+    System.Management.Automation.ErrorRecord
+#>
+function Format-IncorrectValueError([string]$ExpectedKey, [string]$ExpectedValue, [string]$ActualValue) {
+  $Message = "Expected $ExpectedKey to be $ExpectedValue, but got $ActualValue"
+  return Format-ErrorRecord -Message $Message -ErrorID "BenchPressValueFail"
+}
+
+<#
+  .SYNOPSIS
+    Private function to help construct a ErrorRecord.
+
+  .DESCRIPTION
+    Format-ErrorRecord is a private helper function that can be used to construct an ErrorRecord.
+
+  .PARAMETER Message
+    Message for the ErrorRecord
+
+  .PARAMETER ErrorID
+    A string that can be used to uniquily identify the ErrorRecord.
+
+  .EXAMPLE
+    Format-ErrorRecord -Message $incorrectValueMessage -ErrorID "BenchPressValueFail"
+
+  .INPUTS
+    System.String
+
+  .OUTPUTS
+    System.Management.Automation.ErrorRecord
+#>
+function Format-ErrorRecord ([string] $Message, [string]$ErrorID) {
+  $Exception = [Exception] $Message
+  $ErrorCategory = [System.Management.Automation.ErrorCategory]::InvalidResult
+  $TargetObject = @{ Message = $Message }
+  $ErrorRecord = New-Object System.Management.Automation.ErrorRecord $Exception, $ErrorID, $ErrorCategory, $TargetObject
+  return $ErrorRecord
+}
+
+Export-ModuleMember -Function Get-Resource, Get-ResourceByType, Confirm-Resource, `
+  Format-NotExistError, Format-IncorrectValueError, Format-ErrorRecord
```

Comment:
Do we want to export the `Format-*` methods? Wouldn't those be internal helper methods that we don't want to expose outside the Module?

Reply:
They need to be exported in order for unit tests to call these functions. This `Export-ModuleMember` does not expose it to the end user though, only allows for other modules and tests to call them.

Reply:
Gotcha, makes sense.